src/main/java/com/dreammaster/gthandler/multiAirFilter/GT_MetaTileEntity_AirFilterBase.java

```diff
@@ -3,95 +3,135 @@
 import com.dreammaster.gthandler.CustomItemList;
 import com.dreammaster.gthandler.casings.GT_Container_CasingsNH;
 import com.dreammaster.item.ItemList;
+import com.dreammaster.main.MainRegistry;
+import eu.usrv.yamcore.auxiliary.PlayerChatHelper;
 import gregtech.api.enums.Textures;
 import gregtech.api.gui.GT_GUIContainer_MultiMachine;
 import gregtech.api.interfaces.ITexture;
-import gregtech.api.interfaces.metatileentity.IMetaTileEntity;
 import gregtech.api.interfaces.tileentity.IGregTechTileEntity;
 import gregtech.api.items.GT_MetaGenerated_Tool;
 import gregtech.api.metatileentity.implementations.GT_MetaTileEntity_Hatch_Muffler;
 import gregtech.api.metatileentity.implementations.GT_MetaTileEntity_MultiBlockBase;
 import gregtech.api.objects.GT_RenderedTexture;
+import gregtech.api.util.GT_LanguageManager;
 import gregtech.api.util.GT_Multiblock_Tooltip_Builder;
 import gregtech.api.util.GT_Recipe;
 import gregtech.api.util.GT_Utility;
 import gregtech.common.GT_Pollution;
 import gregtech.common.items.GT_MetaGenerated_Tool_01;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.EnumChatFormatting;
 import net.minecraftforge.common.util.ForgeDirection;
+import org.lwjgl.input.Keyboard;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 
-import org.lwjgl.input.Keyboard;
-
-import static gregtech.api.enums.GT_Values.V;
-import static gregtech.api.enums.GT_Values.VN;
+import static gregtech.api.enums.GT_Values.*;
 import static java.lang.Math.max;
 import static java.lang.Math.min;
 
-/**
- * Created by danie_000 on 03.10.2016.
- */
-public class GT_MetaTileEntity_AirFilter extends GT_MetaTileEntity_MultiBlockBase {
+public abstract class GT_MetaTileEntity_AirFilterBase extends GT_MetaTileEntity_MultiBlockBase {
     protected int mPollutionReductionWholeCycle =0;
-    protected int mAccumulatedPollutionReduction =0;
-    protected int baseEff = 2500;
+    protected int baseEff = 0;
+    protected int multiTier = 0;
+    protected int chunkIndex = 0;
     protected boolean hasPollution=false;
-    static final GT_Recipe tRecipe= new GT_Recipe(
+    protected int[][] chunkList; //list of the chunks, each entry contains the x and z coords of the chunks
```

Comment:
Can be named `chunkCoordinates`. Also having a two dim array complicates things, you might be able to use some concrete type here. Probably there is a Vector or Point class somwhere, which has xy fields. If there is none, I would still go with typed approach and create small custom class for holding this data and have it here like like `PollutionChunk[]`